compiler-cli/src/publish.rs

```diff
@@ -155,6 +157,61 @@ fn check_for_name_squatting(package: &Package) -> Result<(), Error> {
     Ok(())
 }
 
+// Checks if publishing packages contain default main functions.
+// Main functions with documentation are considered intentional and allowed.
+fn check_for_default_main(package: &Package) -> Result<(), Error> {
+    let package_name = &package.config.name;
+
+    let has_default_main = package
+        .modules
+        .iter()
+        .flat_map(|m| m.ast.definitions.iter())
+        .filter_map(|d| d.main_function())
+        .any(|main| main.documentation.is_none() && is_default_main(main, package_name));
+
+    if has_default_main {
+        return Err(Error::CannotPublishWithDefaultMain {
+            package_name: package_name.clone(),
+        });
+    }
+
+    Ok(())
+}
+
+fn is_default_main(main: &TypedFunction, package_name: &EcoString) -> bool {
+    if main.body.len() != 1 {
+        return false;
+    }
+
+    let Statement::Expression(expr) = main.body.first() else {
```

Comment:
```suggestion
    let Statement::Expression(expression) = main.body.first() else {
```
No shorthands please